src/components/contextMenu/AposContextMenu.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div class="apos-context-menu" :class="modifiers">
+    <AposContextMenuTip />
+    <ul class="apos-context-menu__items">
+      <AposContextMenuItem 
+        v-for="item in menu" 
+        :key="item.action" 
+        :menuItem="item" 
+        v-on:clicked="itemClicked"
+      />
+    </ul>
+  </div>
+</template>
+
+<script>
+import AposLoading from '../loading/AposLoading.vue';
+import AposContextMenuItem from './AposContextMenuItem.vue';
+import AposContextMenuTip from './AposContextMenuTip.vue';
+
+export default {
+  name: 'AposContextMenu',
+  components: {
+    AposLoading,
+    AposContextMenuItem,
+    AposContextMenuTip
+  },
+  props: {
+    menu: Array,
+    tipAlignment: {
+      type: String,
+      default: 'left'
+    },
+    origin: {
+      type: String,
+      default: 'below'
+    },
+  },
+  computed: {
+    modifiers() {
+      let modifiers = 
+        `apos-context-menu--tip-alignment-${this.tipAlignment} 
+        apos-context-menu--origin-${this.origin}`;
+      return modifiers;
+    }
+  },
+  methods: {
+    itemClicked(action) {
+      console.log(`menu heard ${action} was clicked`);
+      this.$emit('itemClicked', action);
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+  .apos-context-menu {
+    position: relative;
+    display: inline-block;
+    margin: 10px;
+    &:before {
+      content: '';
+      width: 20px;
+      height: 20px;
+    }
+  }
+
+  .apos-context-menu--tip-alignment-left {
+    .apos-context-menu__tab {
```

Comment:
I do think these styles should be on the component they're targeting, generally speaking. It might be weird to fill that component with `.apos-context-menu--` type classes, so instead we pass in `:align="tipAlignment" :origin="origin"` to `AposContextMenuTip` as props, and use those to apply these rule sets.

Reply:
I agree this this